src/figuresClasses.ts

```diff
@@ -1,19 +1,102 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  shape: Shape,
+  color: Color,
 
+  getArea (): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    private sideA: number,
+    private sideB: number,
+    private sideC: number,
+  ) {
+    this.checkProperties();
+  }
+
+  getArea(): number {
+    const semiPer = (this.sideA + this.sideB + this.sideC) / 2;
+    const A = semiPer - this.sideA;
+    const B = semiPer - this.sideB;
+    const C = semiPer - this.sideC;
 
+    return Number(Math.sqrt(semiPer * A * B * C).toFixed(2));
+  }
+
+  checkProperties(): void {
+    const maxSide = Math.max(this.sideA, this.sideB, this.sideC);
+    const twoOtherSides = this.sideA + this.sideB + this.sideC - maxSide;
+
+    if (maxSide >= twoOtherSides) {
+      throw new Error('sides can\'t form a triangle');
+    }
```

Comment:
Don't you check if all sides are more than 0?

Reply:
also agree that need to chech sides